Services/Administration/OJS.Services.Administration.Business/UsersMentors/UsersMentorsBusinessService.cs

```diff
@@ -32,16 +38,24 @@ public async Task<UserMentorAdministrationModel> Create(UserMentorAdministration
         return model;
     }
 
-    public async Task<UserMentorAdministrationModel> Edit(UserMentorAdministrationModel model)
+    public override async Task<UserMentorAdministrationModel> Edit(UserMentorAdministrationModel model)
     {
+        if (model.Id is null)
+        {
+            throw new BusinessServiceException("An invalid id was provided.");
+        }
+
         var userMentor = await this.userMentorData
-            .OneById(model.Id);
+            .GetQuery(us => us.Id == model.Id)
+            .FirstOrDefaultAsync();
 
         if (userMentor is null)
         {
             throw new BusinessServiceException($"The record with id #{model.Id} could not be found.");
         }
 
+        this.userProfileData.Detach(userMentor.User);
```

Comment:
It was added due to a `tracking` error after removing the `UserProfile` entity from the `UserMentorAdministrationModel` the error was fixed, so it can be removed.